### DISCUSSION TITLE
Error: ...already exists in stack arn:aws:cloudformation....

### DISCUSSION THREAD

**Discussion by Air-Craft** [serverless/serverless#10276] [Q&A]
Hi, I'm getting this error below on deploy and I can't understand why there'd be a collision. Shouldnt deploy just update what needs updating?

```
Serverless Error ---------------------------------------

  An error occurred: CognitoTriggersSecurityGroupCI - dev-fp-cognito-triggers already exists in stack arn:aws:cloudformation:us-west-2:366781170337:stack/fp-cognito-triggers-v1.....
```

**Comment by pgrzesik** [comment]
Hello, would you be able to share more details about your setup? Small reproducible example would be the best if possible

**Reply by Air-Craft** [reply]
Thx for replying. There is so much corporate security in the way, I wouldnt even know where to start.  Hold on...it might be a permissions thing....If not I'll try to create a replica on my personal AWS account.

**Reply by Air-Craft** [reply]
I get a bunch of these b4 the error....

```
Serverless Warning --------------------------------------

  A valid SSM parameter to satisfy the declaration 'ssm:/fp-cognito-settings/identity-providers/apple/private-key~true' could not be found.


 Serverless Warning --------------------------------------

  A valid SSM parameter to satisfy the declaration 'ssm:/fp-cognito-settings/identity-providers/apple/private-key~true' could not be found.
```

**Reply by pgrzesik** [reply]
The above would suggest that the settings in SSM are missing - could you confirm that these values in SSM are populated?

**Reply by Air-Craft** [reply]
Yeah I think this is all permissions issues on my side. The key understanding I was lacking is that serverless often creates resources anew and then deletes the old versions, rather than doing a "diff" update

**Reply by pgrzesik** [reply]
Yes, that happens sometimes, that depends on CloudFormation and if changes require replacement of a resource or not

**Reply by Air-Craft** [reply]
It turns out that the service name was based on a command line parameter, which I was passing but not in the correct form and thus it was trying to create a *new* service/app with the same resource names rather than updating the existing one...hence the collisions.

**Reply by pgrzesik** [reply]
Thanks for the update @Air-Craft - glad to hear you've been able to trace the root of the issue 👍
